### PR TITLE
Eliminate UUID keys from entity PUT/POST and template POST in the API.

### PR DESCRIPTION
**Pull request by MTKnife**

### CODE REVIEW

File: src/Coalesce.Services/CRUD/service-data/src/main/java/com/incadencecorp/coalesce/services/crud/service/data/controllers/EntityDataController.java.orig [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    }
+
+    public String getEntityAsXml(String entityKey) throws RemoteException
+    {
+        return retrieveEntity(entityKey).toXml();
+    }
+
+    public void updateEntity(String json) throws RemoteException
+    {
+<<<<<<< HEAD
```
Comment (InCadence) [inline src/Coalesce.Services/CRUD/service-data/src/main/java/com/incadencecorp/coalesce/services/crud/service/data/controllers/EntityDataController.java.orig]: You need to resolve this

File: src/Coalesce.React/common-components/src/js/templateController.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ export function loadTemplateByName(name, source, version)
 
 export function registerTemplate(key)
 {
-  return fetch(`${karafRootAddr}/templates/${key}`, {
+  return fetch(`${karafRootAddr}/templates`, {
```
Comment (InCadence) [inline src/Coalesce.React/common-components/src/js/templateController.js]: This is going to have the same issue as the POST endpoint.

Comment (dclemenzi) [reply src/Coalesce.React/common-components/src/js/templateController.js]: Actually I believe this needs to be reverted. Looking at the API change it still requires a key.

**Comment by dclemenzi** [comment]
OBE

**Pull request closed without merging**